### PR TITLE
docs(v6 migration): document notifications migration path

### PR DESCRIPTION
**Pull request by yairopro**
### Description

Add in the migration documentation a section about the module `notification()` which has been removed from the v6.

### Related issues

None. I just came across this breaking change which wasn't documented ii the "migrating to v6" section.

### Release Summary

Improvement of the "migrating to v6" documentation.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - ✅ No
- My change supports the following platforms;
  - ✅ `Android`
  - ✅ `iOS`
- My change includes tests;
  - [  ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [  ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [  ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - ✅ No

### Test Plan

😩 It's just doc dude.

### CODE REVIEW

**Comment by mikehardy** [comment]
Spell check failure is actually from #3925 

```

12s
##[error]Process completed with exit code 1.
Run spellchecker --quiet --files="docs/**/*.md" --dictionaries="./.spellcheck.dict.txt" --reports="spelling.json" --plugins spell indefinite-article repeated-words syntax-mentions syntax-urls frontmatter

Spellchecking 107 files...

docs/messaging/server-integration.md
  34:79-34:82  warning  `SNS` is misspelt; did you mean `ANS`, `S'S`, `SN'S`, `CNS`, `ENS`, `INS`, `NS`, `SANS`, `SENS`, `SINS`, `SIS`, `SN`, `SONS`, `SOS`, `SS`, `SSS`, `SUNS`?  retext-spell  retext-spell
```

**Comment by Salakar** [comment]
These functions mentioned in this PR aren't, unfortunately, their replacements; `onMessage` is not the same as `onNotification` or `onNotificationOpened`

**Comment by yairopro** [comment]
As I said

> `firebase.notifications().onNotification(Notification => void)` 
> -> `firebase.messaging().onMessage(RemoteMessage => void)`

The functions signatures are indeed different. 
What I was trying to say is that features that have been coded using `notification()` module, can be recoded with the `message()` module using those **similar** functions.

File: docs/migrating-to-v6.md [hunk 1]
```diff
@@ -446,6 +446,16 @@ No breaking changes.
 - `firebase.utils.Native` is now deprecated and will be removed in a later release, please rename usages of this to `firebase.utils.FilePath`.
 - `firebase.utils.Native.*` some properties have been renamed and deprecated and will be removed in a later release, follow the in-app console warnings on how to migrate.
 
+### RN-Firebase notification
+In v5, `firebase.notifications()` was used to listen when a notification received or clicked, or when the app has been opened from a notification.
+This module doesn't exist anymore.
+Here are the other functions to do the same things:
+- `firebase.notifications().onNotification(Notification => void)` -> `firebase.messaging().onMessage(RemoteMessage => void)`.
+- `firebase.notifications().onNotificationOpened(NotificationOpen => void)` -> `firebase.messaging().onMessage(RemoteMessage => void)`.
+- `firebase.notifications().getInitialNotification()` -> `firebase.messaging().getInitialNotification()`.
```
Comment (mikehardy) [inline docs/migrating-to-v6.md:455]:
```suggestion
- `firebase.notifications().getInitialNotification()` -> `firebase.messaging().onNotificationOpenedApp()`.
```

File: docs/migrating-to-v6.md [hunk 2]
```diff
@@ -446,6 +446,16 @@ No breaking changes.
 - `firebase.utils.Native` is now deprecated and will be removed in a later release, please rename usages of this to `firebase.utils.FilePath`.
 - `firebase.utils.Native.*` some properties have been renamed and deprecated and will be removed in a later release, follow the in-app console warnings on how to migrate.
 
+### RN-Firebase notification
+In v5, `firebase.notifications()` was used to listen when a notification received or clicked, or when the app has been opened from a notification.
+This module doesn't exist anymore.
+Here are the other functions to do the same things:
+- `firebase.notifications().onNotification(Notification => void)` -> `firebase.messaging().onMessage(RemoteMessage => void)`.
+- `firebase.notifications().onNotificationOpened(NotificationOpen => void)` -> `firebase.messaging().onMessage(RemoteMessage => void)`.
```
Comment (mikehardy) [inline docs/migrating-to-v6.md:454]:
```suggestion
- `firebase.notifications().onNotification(Notification => void)` -> Use a 3rd party notifications module API.
- `firebase.notifications().onNotificationOpened(NotificationOpen => void)` -> Use a 3rd party notifications module API.
```

**Comment by mikehardy** [comment]
@yairopro you have not signed the CLA, I patched up the PR to match desired wording but you must sign the CLA in order for us to merge

**Pull request closed without merging**